# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/loader/WorkerModuleLoader.h
# Commit: 474668f6615e
# Full Hash: 474668f6615eafb7bc9a94b5bcf6aada66ea5d99
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2023-01-19 04:46:07
# Regressor Bug: 1247687
# File Overlap Count: 1
# Description:
#   Bug 1247687 - Implement initial Static Module Loading for Workers; r=jonco
#   
#   This is the first pass of getting static module loading to work. This roughly implements
#   https://html.spec.whatwg.org/multipage/webappapis.html#fetch-a-worklet/module-worker-script-graph --
#   without some of the settings objects correctly set.
# ==============================================================================

diff -r 15eb9daca661 -r 474668f6615e dom/workers/loader/WorkerModuleLoader.h
--- a/dom/workers/loader/WorkerModuleLoader.h	Wed Jan 18 13:46:30 2023 +0000
+++ b/dom/workers/loader/WorkerModuleLoader.h	Wed Jan 18 13:46:30 2023 +0000
@@ -19,6 +19,16 @@
 using ScriptLoadRequestList = JS::loader::ScriptLoadRequestList;
 using ModuleLoadRequest = JS::loader::ModuleLoadRequest;
 
+// WorkerModuleLoader
+//
+// The WorkerModuleLoader provides the methods that implement specification
+// step 5 from "To fetch a worklet/module worker script graph", specifically for
+// workers. In addition, this implements worker specific initialization for
+// Static imports and Dynamic imports.
+//
+// The steps are outlined in "To fetch the descendants of and link a module
+// script" and are common for all Modules. Thus we delegate to ModuleLoaderBase
+// for those steps.
 class WorkerModuleLoader : public JS::loader::ModuleLoaderBase {
  public:
   NS_DECL_ISUPPORTS_INHERITED
@@ -45,6 +55,8 @@
 
   bool CanStartLoad(ModuleLoadRequest* aRequest, nsresult* aRvOut) override;
 
+  // StartFetch is special for worker modules, as we need to move back to the
+  // main thread to start a new load.
   nsresult StartFetch(ModuleLoadRequest* aRequest) override;
 
   nsresult CompileFetchedModule(
